spec/models/mail_server_log_spec.rb

```diff
@@ -331,21 +331,56 @@
 
   describe '#delivery_status' do
 
-    context 'using the :exim MTA' do
+    context 'if there is a stored value' do
+      let(:log) do
+        MailServerLog.new(:line => "log text",
+                          :delivery_status => 'bounce_arrival')
+      end
+
+      it 'returns the stored value' do
+        status = MailServerLog::EximDeliveryStatus.new(:bounce_arrival)
+        expect(log.delivery_status).to eq(status)
+      end
+
+      it 'does not look at the line text' do
+        expect(log).to_not receive(:line)
+        log.delivery_status
+      end
+
+    end
+
+    context 'there is not a stored value' do
+
+      it 'attempt to parse the line text' do
+        log = MailServerLog.new(:line => "hi")
+        expect(log).to receive(:line)
+        log.delivery_status
+      end
+
+      context 'using the :exim MTA' do
+        let(:line) do
+          "Apr 28 15:53:37 server exim[12105]: 2016-04-28 15:53:37 " \
+          "[12105] 1avnJx-00039F-Hs <= " \
+          "foi+request-331612-13811a2b@example.com U=foi P=local " \
+          "S=1986 id=ogm-538593+572f16e888-166a@example.com " \
+          "T=\"Freedom of Information request - example request\" " \
+          "from <foi+request-331612-13811a2b@example.com> for " \
+          "foi@example.org foi@example.org"
+        end
 
         it 'returns a delivery status for the log line' do
-          line = "Apr 28 15:53:37 server exim[12105]: 2016-04-28 15:53:37 " \
-               "[12105] 1avnJx-00039F-Hs <= " \
-               "foi+request-331612-13811a2b@example.com U=foi P=local " \
-               "S=1986 id=ogm-538593+572f16e888-166a@example.com " \
-               "T=\"Freedom of Information request - example request\" " \
-               "from <foi+request-331612-13811a2b@example.com> for " \
-               "foi@example.org foi@example.org"
-          log = MailServerLog.new(:line => line)
+          log = MailServerLog.new(:line => line, :order => 1)
```

Comment:
a side effect of the `save` operation - it chokes without it, might be able to get rid of this post refactoring